File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
#include<map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        if (size(nums) == 2) return {0,1};
        std::map<int, int> m;
        for (int i = 0; i < size(nums); ++i){
            if (!m.contains(target-nums[i])) {
                m[nums[i]] = i;
            }
            else return {m[target-nums[i]], i};
        }
    }
};

File: Data Structures & Algorithms/permutations/submission-3.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        perm = [[]]
        for i in range(len(nums)):
            tmp = []
            for p in perm: tmp += [p[:j] + [nums[i]] + p[j:] for j in range(i+1)]
            perm = tmp

        return perm
